### PR TITLE
Add support for external slurmdbd in Slurm configuration

### PR DESCRIPTION
**Pull request by jdeamicis**
### Description of changes
* Use ExternalSlurmdbd to render AccountingStorageHost when defined.

### Tests
* Add unit test case for Slurm config rendering to cover external slurmdbd scenario.
* Adapt other unit tests for Slurm configuration rendering when Slurm Accounting is used.
* Successful manual test to connect a cluster to a prototype external slurmdbd stack.

### References
* https://github.com/aws/aws-parallelcluster/pull/5926

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/unit/slurm/test_slurm_config_generator/test_generate_slurm_config_files_slurm_accounting/expected_outputs/slurm_parallelcluster_external_slurmdbd.conf [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+SuspendTime=600
+ResumeTimeout=1600
+SelectTypeParameters=CR_CPU
+AccountingStorageType=accounting_storage/slurmdbd
+AccountingStorageHost=test.slurmdbd.host
+AccountingStoragePort=6819
+AccountingStorageUser=slurm
+JobAcctGatherType=jobacct_gather/cgroup
+
+include <DIR>/pcluster/slurm_parallelcluster_efa_partition.conf
```
Comment (NSsirena) [inline test/unit/slurm/test_slurm_config_generator/test_generate_slurm_config_files_slurm_accounting/expected_outputs/slurm_parallelcluster_external_slurmdbd.conf]: Why we need to include efa configuration in the slurm_dbd.conf ?

Comment (jdeamicis) [reply test/unit/slurm/test_slurm_config_generator/test_generate_slurm_config_files_slurm_accounting/expected_outputs/slurm_parallelcluster_external_slurmdbd.conf]: This is not the `slurmdbd.conf`, this is still the `slurm.conf`. But I see that the name of the file can be misleading (here I'm using `external_slurmdbd` as an appendix for the files in the external dbd case).
What appendix should we use here? `externaldbd`?

Comment (NSsirena) [reply test/unit/slurm/test_slurm_config_generator/test_generate_slurm_config_files_slurm_accounting/expected_outputs/slurm_parallelcluster_external_slurmdbd.conf]: Ahh.. ok. Yes, that is misleading.
Let's try with "_externaldbd"

Comment (jdeamicis) [reply test/unit/slurm/test_slurm_config_generator/test_generate_slurm_config_files_slurm_accounting/expected_outputs/slurm_parallelcluster_external_slurmdbd.conf]: Done. Resolving.

**Review by NSsirena — APPROVED** [review]

**Pull request merged**